Mathlib/NumberTheory/Cyclotomic/PrimitiveRoots.lean

```diff
@@ -461,47 +461,47 @@ theorem pow_sub_one_norm_prime_pow_ne_two {k s : ℕ} (hζ : IsPrimitiveRoot ζ
     norm K (ζ ^ (p : ℕ) ^ s - 1) = (p : K) ^ (p : ℕ) ^ s := by
-  refine' hζ.pow_sub_one_norm_prime_pow_ne_two hirr hs fun h => _
+  refine' hζ.norm_pow_sub_one_of_prime_pow_ne_two hirr hs fun h => _
   have coe_two : ((2 : ℕ+) : ℕ) = 2 := by norm_cast
   rw [← PNat.coe_inj, coe_two, PNat.pow_coe, ← pow_one 2] at h
 -- Porting note: the proof is slightly different because of coercions.
   replace h :=
     eq_of_prime_pow_eq (prime_iff.1 hpri.out) (prime_iff.1 Nat.prime_two) (k - s).succ_pos h
   exact hodd (PNat.coe_injective h)
 
-#align is_primitive_root.pow_sub_one_norm_prime_ne_two IsPrimitiveRoot.pow_sub_one_norm_prime_ne_two
+#align is_primitive_root.pow_sub_one_norm_prime_ne_two IsPrimitiveRoot.norm_pow_sub_one_of_prime_ne_two
 
 /-- If `Irreducible (cyclotomic (p ^ (k + 1)) K)` (in particular for `K = ℚ`) and `p` is an odd
 prime, then the norm of `ζ - 1` is `p`. -/
-theorem sub_one_norm_prime_ne_two {k : ℕ} (hζ : IsPrimitiveRoot ζ ↑(p ^ (k + 1)))
+theorem norm_sub_one_of_prime_ne_two {k : ℕ} (hζ : IsPrimitiveRoot ζ ↑(p ^ (k + 1)))
     [hpri : Fact (p : ℕ).Prime] [IsCyclotomicExtension {p ^ (k + 1)} K L]
     (hirr : Irreducible (cyclotomic (↑(p ^ (k + 1)) : ℕ) K)) (h : p ≠ 2) : norm K (ζ - 1) = p := by
-  simpa using hζ.pow_sub_one_norm_prime_ne_two hirr k.zero_le h
-#align is_primitive_root.sub_one_norm_prime_ne_two IsPrimitiveRoot.sub_one_norm_prime_ne_two
+  simpa using hζ.norm_pow_sub_one_of_prime_ne_two hirr k.zero_le h
+#align is_primitive_root.sub_one_norm_prime_ne_two IsPrimitiveRoot.norm_sub_one_of_prime_ne_two
 
 /-- If `Irreducible (cyclotomic p K)` (in particular for `K = ℚ`) and `p` is an odd prime,
 then the norm of `ζ - 1` is `p`. -/
-theorem sub_one_norm_prime [hpri : Fact (p : ℕ).Prime] [hcyc : IsCyclotomicExtension {p} K L]
-    (hζ : IsPrimitiveRoot ζ p) (hirr : Irreducible (cyclotomic p K)) (h : p ≠ 2) :
-    norm K (ζ - 1) = p := by
+theorem norm_sub_one_of_prime_ne_two' [hpri : Fact (p : ℕ).Prime]
+    [hcyc : IsCyclotomicExtension {p} K L] (hζ : IsPrimitiveRoot ζ p)
+    (hirr : Irreducible (cyclotomic p K)) (h : p ≠ 2) : norm K (ζ - 1) = p := by
   replace hirr : Irreducible (cyclotomic (p ^ (0 + 1) : ℕ) K) := by simp [hirr]
   replace hζ : IsPrimitiveRoot ζ (p ^ (0 + 1) : ℕ) := by simp [hζ]
   haveI : IsCyclotomicExtension {p ^ (0 + 1)} K L := by simp [hcyc]
-  simpa using sub_one_norm_prime_ne_two hζ hirr h
-#align is_primitive_root.sub_one_norm_prime IsPrimitiveRoot.sub_one_norm_prime
+  simpa using norm_sub_one_of_prime_ne_two hζ hirr h
+#align is_primitive_root.sub_one_norm_prime IsPrimitiveRoot.norm_sub_one_of_prime_ne_two'
 
 /-- If `Irreducible (cyclotomic (2 ^ (k + 1)) K)` (in particular for `K = ℚ`), then the norm of
 `ζ ^ (2 ^ k) - 1` is `(-2) ^ (2 ^ k)`. -/
 -- Porting note: writing `(2 : ℕ+)` was not needed (similarly everywhere).
-theorem pow_sub_one_norm_two {k : ℕ} (hζ : IsPrimitiveRoot ζ (2 ^ (k + 1)))
+theorem norm_pow_sub_one_of_eq_two {k : ℕ} (hζ : IsPrimitiveRoot ζ (2 ^ (k + 1)))
```

Comment:
I guess this is for consistency with the `prime_ne_two` names, but `of_eq_two` doesn't really make sense

Reply:
Ah yes, let's keep `two`.